### ISSUE TITLE
PowerShell Extension terminal hangs when using Windows PowerShell

### ISSUE THREAD

**Issue by sba923** [PowerShell/vscode-powershell#4731]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [_open and closed_ issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.
- [X] I have read the [troubleshooting](https://github.com/PowerShell/vscode-powershell/blob/main/docs/troubleshooting.md) guide.
- [X] I am sure this issue is with the _extension itself_ and does not reproduce in a standalone [PowerShell](https://github.com/PowerShell/PowerShell/issues/new/choose) instance.
- [X] I have verified that I am using the latest version of Visual Studio Code and the PowerShell extension.
- [X] If this is a security issue, I have read the [security issue reporting guidance](https://github.com/PowerShell/vscode-powershell/blob/main/SECURITY.md).

### Summary

While investigating another issue I switched to "Windows PowerShell (x64)" in the "PowerShell Session Menu."

Now the session doesn't start anymore, it hangs at:

````
PowerShell Extension v2023.6.0
Copyright (c) Microsoft Corporation.

https://aka.ms/vscode-powershell
Type 'help' to get help.
````

### PowerShell Version

```console
Name                           Value
----                           -----
PSVersion                      5.1.19041.3031
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.3031
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

(assumed, given that I can't get that PowerShell version to run within VScode)
```

### Visual Studio Code Version

```console
1.81.1
6c3e3dba23e8fadc360aed75ce363ba185c49794
x64
```

### Extension Version

```console
ms-vscode.powershell@2023.6.0
TylerLeonhardt.vscode-inline-values-powershell@0.0.7
```

### Steps to Reproduce

1. switch to "Windows PowerShell (x64)" via the PowerShell Session Menu (resulting in `powershell.powerShellDefaultVersion` being set to `Windows PowerShell (x64)`.
2. A "session restart" occurs
3. The session never finishes initializing

When `powershell.powerShellDefaultVersion` is set to `PowerShell (x64)` the problem goes away.

(I know support for Windows PowerShell 5.1 is "best effort")

### Logs

[169408~1.ZIP](https://github.com/PowerShell/vscode-powershell/files/12549014/169408.1.ZIP)

**Comment by SydneyhSmith** [comment]
Thanks @sba923 it looks like it having issues at the loading profile step, do you have anything in your profile? does it work without a profile?

**Comment by sba923** [comment]
My `$profile.CurrentUserAllHosts` contains one line, sourcing my actual `profile.ps1` that resides elsewhere (and is 1,004 lines long 🙃).

If that comment out that line, the problem remains.

**Comment by sba923** [comment]
I can repro this on another machine...

**Comment by andyleejordan** [comment]
Have you tried with `powershell.enableProfileLoading` to _completely_ rule it out?

**Comment by sba923** [comment]
Not yet... but... I hate to say... I can't seem to repro the issue right now... I'll keep trying...

**Comment by sba923** [comment]
Not yet... but... I hate to say... I can't seem to repro the issue right now... I'll keep trying...

**Comment by sba923** [comment]
I confirm the extension doesn't hang anymore, but it's unusable due to something that looks related to `PSReadLine`:

![image](https://github.com/PowerShell/vscode-powershell/assets/12860484/ad697aa5-1995-4782-92f2-b67e497a2aa9)

**Comment by andyleejordan** [comment]
What custom key handlers do you have set, can you run it without them?

**Comment by sba923** [comment]
I've found the culprit, and it's not `PSReadLine`-related.

Wrapping the loading of `oh-my-posh` within

```powershell
if (($PSVersionTable.PSVersion.Major -ne 5) -or ($host.Name -notmatch 'Visual Studio Code Host'))
{
[...]
}
```

gives me a working session.

I'll ping @JanDeDobbeleer about this.

**Comment by JanDeDobbeleer** [comment]
@sba923 last time I checked 5 worked, but also on my end that's "best effort". It's legacy 😅

**Comment by sba923** [comment]
> @sba923 last time I checked 5 worked, but also on my end that's "best effort". It's legacy 😅

I perfectly. Note that outside of the VScode PowerShell Extension, `oh-my-posh` works like a charm with Windows PowerShell 5.1.

I don't _want_ to support Windows PowerShell in whatever I'm implementing. I _have_ to use it wherever PowerShell 7.x can't be used. And I thus need to debug those scripts...

**Comment by JanDeDobbeleer** [comment]
@sba923 can you get me the content of `$error`? Pretty sure this is related to not having any console properties defined when in this extension's shell runtime.

**Comment by sba923** [comment]
How do you suggest to capture it? As soon as `oh-my-posh` loads, the prompt becomes unusable...

**Comment by JanDeDobbeleer** [comment]
@sba923 can you try again using https://github.com/JanDeDobbeleer/oh-my-posh/releases/tag/v18.21.1?

**Comment by ryan-beesley** [comment]
I think there must be more to this. I have oh-my-posh, but only for PowerShell Core 7.3.9. Only `$PROFILE.CurrentUserCurrentHost` exists and it only has an extension for `$env:Path` to include some scripts I keep available. These scripts are not loaded at all, only the Path environment variable is modified. The version of PSReadLine I'm using in Windows PowerShell is 2.0.0. I'm set `"powershell.enableProfileLoading": false`, and it did not resolve my problem.

```
Version: 1.84.2 (system setup)
Commit: 1a5daa3a0231a0fbba4f14db7ec463cf99d7768e
Date: 2023-11-09T10:51:52.184Z
Electron: 25.9.2
ElectronBuildId: 24603566
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.22000
```

```
PS C:\Users\RyanBeesley> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.22000.2538
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.22000.2538
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1

PS C:\Users\RyanBeesley> Get-Module

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Manifest   3.1.0.0    Microsoft.PowerShell.Management     {Add-Computer, Add-Content, Checkpo...
Script     2.0.0      PSReadline                          {Get-PSReadLineKeyHandler, Get-PSRe...


PS C:\Users\RyanBeesley>
```

**Comment by sba923** [comment]
@ryan-beesley  are you still able to repro this?

As far as I'm concerned, I've just disabled `oh-my-posh` when running Windows PowerShell with the VScode extension.

(@JanDeDobbeleer the problem's still present with 18.25.1)

**Comment by JanDeDobbeleer** [comment]
@sba923 this should be fixed here. It fails in this handler: https://github.com/JanDeDobbeleer/oh-my-posh/blob/b31f01a1a3a127a721e4cf0b5afef9c91f462203/src/shell/scripts/omp.ps1#L212 I can add a try-catch there as well but in my opinion, the behaviour should be identical here as when running native pwsh.

**Comment by sba923** [comment]
> @sba923 this should be fixed here. It fails in this handler: https://github.com/JanDeDobbeleer/oh-my-posh/blob/b31f01a1a3a127a721e4cf0b5afef9c91f462203/src/shell/scripts/omp.ps1#L212 I can extend the try-catch there as well but in my opinion, the behaviour should be identical here as when running native pwsh.

I hate to say, but that's clearly not the case:

![image](https://github.com/PowerShell/vscode-powershell/assets/12860484/bd6bfef2-4e64-42c8-9cf2-c15c36bb0459)

**Comment by JanDeDobbeleer** [comment]
> I hate to say, but that's clearly not the case:

What do you mean?

**Comment by sba923** [comment]
> > I hate to say, but that's clearly not the case:
> 
> What do you mean?

I mean the issue is still present with `oh-my-posh` 18.26.1.

What did you mean by "it should be fixed _here_"?

**Comment by JanDeDobbeleer** [comment]
@sba923 this repository.

**Comment by sba923** [comment]
> @sba923 this repository.

Oh, I see... need to find out how to get the issue reopened...

**Comment by ryan-beesley** [comment]
@sba923, ~~yes, this is still an issue for me. I resorted to using Windows PowerShell ISE to edit and debug my script for Windows PowerShell 5.1, so while this is a workaround the issue raised by this work item still applies to me and isn't seemingly related to oh-my-posh. I have oh-my-posh installed, but this problem remains even when I have profile loading disabled.~~

I just went through and updated everything (Oh-My-Posh, PowerShell 7.4.0), again, and restarted VSCode. The extension is now working with Windows PowerShell 5.1, so this has been resolved now for me. Neither update seems as though it should have had an impact, because the 5.1 profile wasn't being loaded, only `$PROFILE.CurrentUserCurrentHost` exists for both Windows PowerShell 5.1 and PowerShell Core 7.4.0, different scripts, and with regard to Windows PowerShell `$PROFILE.CurrentUserCurrentHost` only adds a directory to the `$env:Path` so Oh-My-Posh wasn't being initiated at all. I'm not sure what changed to make it work.
